src/http-tequilapi-client.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2019 The "mysteriumnetwork/js-tequilapi" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { HttpInterface } from './adapters/interface'
+import ProposalsQuery from './adapters/proposals-query'
+import { TequilapiClient } from './client'
+import { ConnectionIPDTO, parseConnectionIPDTO } from './dto/connection-ip'
+import { ConnectionStatisticsDTO, parseConnectionStatisticsDTO } from './dto/connection-statistics'
+import { ConnectionStatusDTO, parseConnectionStatusDTO } from './dto/connection-status-dto'
+import { ConsumerLocationDTO, parseConsumerLocationDTO } from './dto/consumer-location'
+import { parseIdentitiesResponseDTO } from './dto/identities-response'
+import { IdentityDTO, parseIdentityDTO } from './dto/identity'
+import {
+  IdentityRegistrationDTO,
+  parseIdentityRegistrationDTO
+} from './dto/identity-registration/identity-registration'
+import { NodeHealthcheckDTO, parseHealthcheckResponse } from './dto/node-healthcheck'
+import { ProposalDTO } from './dto/proposal'
+import { parseProposalsResponseDTO } from './dto/proposals-response'
+import { ConnectionRequest } from './dto/query/connection-request'
+import { ProposalQueryOptions } from './dto/query/proposals-query-options'
+import { TIMEOUT_DISABLED } from './timeouts'
+
+export class HttpTequilapiClient implements TequilapiClient {
+  public http: HttpInterface
+
+  constructor (http: HttpInterface) {
+    this.http = http
+  }
+
+  public async healthCheck (timeout?: number): Promise<NodeHealthcheckDTO> {
+    const response = await this.http.get('healthcheck', undefined, timeout)
+    if (!response) {
+      throw new Error('Healthcheck response body is missing')
+    }
+    return parseHealthcheckResponse(response)
+  }
+
+  public async stop (): Promise<void> {
+    await this.http.post('stop')
+  }
+
+  public async identitiesList (): Promise<IdentityDTO[]> {
+    const response = await this.http.get('identities')
+    if (!response) {
+      throw new Error('Identities response body is missing')
+    }
+    const responseDto = parseIdentitiesResponseDTO(response)
+
+    return responseDto.identities
+  }
+
+  public async identityCreate (passphrase: string): Promise<IdentityDTO> {
+    const response = await this.http.post('identities', { passphrase })
+    if (!response) {
```

Comment:
feels like you could wrap the response + this if block into a separate function and just call that everywhere instead of repeating it in every method.

Reply:
You mean something like this?

```typescript
private async getResponse (path: string, errorMessage: string, query?: HttpQueryParams, timeout?: number): Promise<any> {
  const response = await this.http.get(path, query, timeout)
  if (!response) {
    throw new Error(errorMessage)
  }
  return response
}

// same for post
// same for delete
// same for put
```

Reply:
yup